feat: refactor and add protocol support to `stats/base/nanrange-by`

Kaushikgtm opened the pull request:
Resolves #5666.

## Description

> What is the purpose of this pull request?

This pull request:

-   {{TODO: }

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5666  

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Kaushikgtm committed:
Create accessors.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update accessors.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update nanrange_by.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update ndarray.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update test.nanrange_by.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update test.ndarray.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update index.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update index.d.ts

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update test.ts

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update repl.txt

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update benchmark.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update benchmark.ndarray.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update accessors.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update ndarray.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update accessors.js

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

Kaushikgtm committed:
Update README.md

Signed-off-by: Kaushikgtm <162317291+Kaushikgtm@users.noreply.github.com>

ShabiShett07 commented on lib/node_modules/@stdlib/stats/base/nanrange-by/README.md:
````diff
@@ -185,22 +186,23 @@ var v = nanrangeBy.ndarray( 3, x, 1, x.length-3, accessor );
 
 ```javascript
 var discreteUniform = require( '@stdlib/random/base/discrete-uniform' );
````
```suggestion
```

ShabiShett07 replied:
This is an unused package so no need of defining it.

kgryte commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Kaushikgtm-patch-nanrange-by

gururaj1512 commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into Kaushikgtm-patch-nanrange-by

gururaj1512 committed:
fix: readme file

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: update repl

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: update type definitions for nanrangeBy

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: correct test.ts

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: examples

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: implementation

gururaj1512 committed:
chore: minor cleanup

gururaj1512 committed:
fix: test files

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

gururaj1512 committed:
fix: ensure original input array is passed to callback

Signed-off-by: Gururaj Gurram <143020143+gururaj1512@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanrange-by/lib/main.js:
```diff
@@ -20,14 +20,35 @@
+* // returns 18.0
+*/
+function nanrangeBy( N, x, strideX, clbk, thisArg ) {
+	return ndarray( N, x, strideX, stride2offset( N, strideX ), clbk, thisArg);
```
```suggestion
	return ndarray( N, x, strideX, stride2offset( N, strideX ), clbk, thisArg );
```

kgryte committed:
style: fix missing space

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanrange-by/test/test.main.js:
```diff
@@ -75,18 +75,54 @@ tape( 'the function calculates the range of a strided array via a callback funct
 	t.end();
 });
 
+tape( 'the function calculates the range of a strided array via a callback function, ignoring NaN values (accessor)', function test( t ) {
```
```suggestion
tape( 'the function calculates the range of a strided array via a callback function, ignoring NaN values (accessors)', function test( t ) {
```

kgryte committed:
test: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanrange-by/test/test.ndarray.js:
```diff
@@ -101,6 +137,21 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
 	t.end();
 });
 
+tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN` (accessor)', function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter less than or equal to `0`, the function returns `NaN` (accessors)', function test( t ) {
```

kgryte committed:
test: update desc

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanrange-by/test/test.ndarray.js:
```diff
@@ -110,15 +161,30 @@ tape( 'if provided an `N` parameter equal to `1`, the function returns `0`', fun
 	t.end();
 });
 
+tape( 'if provided an `N` parameter equal to `1`, the function returns `0` (accessor)', function test( t ) {
```
```suggestion
tape( 'if provided an `N` parameter equal to `1`, the function returns `0` (accessors)', function test( t ) {
```

kgryte committed:
test: update desc

Signed-off-by: Athan <kgryte@gmail.com>
